openvdb_points/tools/IndexFilter.h

```diff
@@ -362,51 +366,57 @@ class LevelSetFilter
         const openvdb::Vec3f& pointVoxelSpace = mPositionHandle->get(*iter);
 
         // Compute point position in index space
-        const openvdb::Vec3f pointWorldSpace = mData->transform.indexToWorld(pointVoxelSpace + voxelIndexSpace);
-        const openvdb::Vec3f pointIndexSpace = mData->levelSetTransform.worldToIndex(pointWorldSpace);
+        const openvdb::Vec3f pointWorldSpace = mTransform.indexToWorld(pointVoxelSpace + voxelIndexSpace);
+        const openvdb::Vec3f pointIndexSpace = mLevelSetTransform.worldToIndex(pointWorldSpace);
 
         // Perform level-set sampling
-        const typename LevelSetGridT::ValueType value = BoxSampler::sample(mData->accessor, pointIndexSpace);
+        const typename LevelSetGridT::ValueType value = BoxSampler::sample(mAccessor, pointIndexSpace);
 
         // if min is greater than max, we invert so that values are valid outside of the range (not inside)
-        const bool invert = mData->min > mData->max;
+        const bool invert = mMin > mMax;
 
-        return invert ? (value < mData->max || value > mData->min) : (value < mData->max && value > mData->min);
+        return invert ? (value < mMax || value > mMin) : (value < mMax && value > mMin);
     }
 
 private:
-    const Data* mData;
-    AttributeHandle<openvdb::Vec3f>::Ptr mPositionHandle;
+    // not a reference to ensure const-accessor is unique per-thread
+    const typename LevelSetGridT::ConstAccessor mAccessor;
+    const math::Transform& mLevelSetTransform;
+    const math::Transform& mTransform;
+    const ValueT mMin;
+    const ValueT mMax;
+    Handle::ScopedPtr mPositionHandle;
 }; // class LevelSetFilter
 
 
 // BBox index filtering
 class BBoxFilter
 {
 public:
-    struct Data
+    typedef AttributeHandle<openvdb::Vec3f> Handle;
```

Comment:
In looking through this, it occurred to me that we have several instances of this typedef throughout the code base purely for accessing the position attribute. Given that position is built-in, would it be worth introducing a `PositionHandle` and `PositionWriteHandle` typedef at the library-level to reduce this duplication?